src/services/ExplorerService/hooks.ts

```diff
@@ -0,0 +1,10 @@
+import { useObservableState } from 'observable-hooks'
+import { explorerService } from '.'
+
+export function useStatistics() {
+  return useObservableState(explorerService.latestStatistics$)
```

Comment:
What's the difference between using `useObservableState` and `useQuery`

Reply:
useQuery does not support Observable type inputs unless we wrap it ourselves, but implementing this has many pitfalls.

Reply:
I'm curious about the effect of using `useObservableState` and `useQuery`.

I found `useObservableState` was used at https://github.com/Magickbase/ckb-explorer-frontend/pull/103/files#diff-6fcb1d89436736d15428c53cb68462b5cc1fdffbcca5fc31f716aea1c5b5a3c6R19

```js
  const { reorgStartedAt } = useStatistics()
```

And it can be done by `useQuery` as follows

```js
const { data: { reorg_started_at } } = useQuery(api_url)
```

and `reorg_started_at` is sharable because `useQuery` caches it.

Do they have different effects?

Reply:
> ```js
> const { data: { reorg_started_at } } = useQuery(api_url)
> ```

Oh, I understand what you mean now.

ExplorerService was created to aggregate all the code that interacts with the Explorer backend in one place. Currently, there is still a lot of code scattered throughout the project (such as fetcher.ts), but I expect to refactor them into this service in the future.

This ExplorerService will be responsible for data communication, data model conversion, and some persistence logic between the project itself and the external service of the explorer backend, and it belongs to the glue layer. Everything that the explorer backend can provide should be converted into internal models within the front-end project through this layer.

This can separate the part of the front-end business that communicates with the external service, making the business abstraction clearer and easier to maintain.

For an extreme example, if we want to replace React with Vue, we only need to modify the front-end business layer, without touching the service-related code.

Of course, it is also feasible for ExplorerService to provide fetchStatistics externally, but I feel that periodic requests for statistics can also fall within the scope of services that ExplorerService can provide.

Reply:
Make sense.